tools/Makefile

```diff
@@ -151,15 +165,36 @@ endif
 ifeq ($(HOST_OS),Darwin)
   tools-y += bash
 else
-  $(curdir)/dwarves/compile += $(curdir)/elfutils/compile
+  $(curdir)/dwarves/compile += $(curdir)/cmake/compile $(curdir)/elfutils/compile
   tools-$(if $(CONFIG_BUILD_ALL_HOST_TOOLS)$(CONFIG_DWARVES),y) += dwarves
 endif
 
 ifneq ($(CONFIG_CCACHE)$(CONFIG_SDK),)
-$(foreach tool, $(filter-out zstd zlib xz pkgconf patch ninja meson libressl expat cmake,$(tools-y)), $(eval $(curdir)/$(tool)/compile += $(curdir)/ccache/compile))
-tools-y += ccache xxhash
-$(curdir)/xxhash/compile := $(curdir)/cmake/compile
-$(curdir)/ccache/compile := $(curdir)/xxhash/compile
+  tools-noccache :=
+  tools-noccache += autoconf
+  tools-noccache += automake
+  tools-noccache += cmake
+  tools-noccache += e2fsprogs
+  tools-noccache += expat
+  tools-noccache += gnulib
+  tools-noccache += libressl
+  tools-noccache += libtool
+  tools-noccache += lzma
+  tools-noccache += m4
+  tools-noccache += meson
```

Comment:
why is meson here?

Reply:
good point let me check which ones are really needed now

Reply:
meson might as well stay in the list because it's not even a compilation right?